src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -305,6 +318,40 @@ def validate(self, data):
                     has_query_error = True
                 else:
                     query_errors.append({})
+
+                if not query.get("on_demand_extraction_disabled", False):
```

Comment:
Let's add a flag check w/ `organizations:on-demand-metrics-extraction-widgets` in addition to the passed `disabled` flag here so we're not making these for orgs that aren't extracting. If it's not set, all the specs will be wrong anyway because of https://github.com/getsentry/sentry/blob/7f3992553de326213c080a68f22033940a193ab7/src/sentry/relay/config/metric_extraction.py#L197